pip/utils/encoding.py

```diff
@@ -20,4 +23,9 @@ def auto_decode(data):
     for bom, encoding in BOMS:
         if data.startswith(bom):
             return data[len(bom):].decode(encoding)
+    # Lets check the first two lines as in PEP263
+    for line in data.splitlines()[:2]:
+        if line.startswith('#') and ENCODING_RE.search(line):
```

Comment:
Huh? I thought I added a line comment here. Trying again.

```data``` is a bytes object. In Python 3, it looks like bytes objects have a ```splitlines``` method, but it behaves very weirdly:

```
>>> u'a\nb£'.encode('utf-8').splitlines()
['a', 'b\xc2\xa3']
```

Note that splitlines returns (unicode) string objects with encoded byte values in them! I've submitted http://bugs.python.org/issue26501 for this problem, but we should not be using splitlines on bytes objects.

Looks like the tests on Python 3 are failing for this...